src/utils/__tests__/initilizerTests.js

```diff
@@ -0,0 +1,388 @@
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(res.reduxMiddleware, mw);
+});
+
+test('init sets context.components as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { components: { Plugin: 0, User: false } },
+        { components: { Plugin: 1 } },
+      ],
+      components: { User: true },
+    },
+  };
+  const defaults = {
+    components: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.components, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.settingsComponentObjects as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { settingsComponentObjects: { Plugin: 0, User: false } },
+        { settingsComponentObjects: { Plugin: 1 } },
+      ],
+      settingsComponentObjects: { User: true },
+    },
+  };
+  const defaults = {
+    settingsComponentObjects: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.settingsComponentObjects, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.events as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { events: { Plugin: 0, User: false } },
+        { events: { Plugin: 1 } },
+      ],
+      events: { User: true, User2: true },
+    },
+  };
+  const defaults = {};
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.events, {
+    Plugin: 1,
+    User: false, // TODO: bug that plugins overwrite user events?
+    User2: true,
+  });
+});
+
+test('init sets context.selectors as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { selectors: { Plugin: 0 } },
+        { selectors: { Plugin: 1 } },
+      ],
+    },
+  };
+  const defaults = {
+    selectors: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.selectors, {
+    Defaults: true,
+    Plugin: 1,
+  });
+});
+
+
+test('init sets context.listeners as expected given props (plugins, user)', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { listeners: { plugin: () => 0, user: () => false } },
+        { listeners: { plugin: () => 1 } },
+      ],
+      listeners: {
+        user: () => true,
+        user2: () => true,
+      },
+    },
+  };
+  const defaults = {};
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+  assert.truthy(res);
+
+  assert.false('defaults' in ctx.listeners);
+  assert.deepEqual(ctx.listeners.plugin(), 1);
+  assert.deepEqual(ctx.listeners.user(), false); // TODO: bug that plugins overwrite user listeners?
```

Comment:
@joellanciaux @ryanlanciaux again: should `<Griddle listeners={...}>` be overwritten by plugins?

cc @shorja